eos-tech-support/eos-write-image

```diff
@@ -36,7 +36,7 @@ EOF
 FORCE=
 REMOVABLE=
 DD_IN_OPTS=(iflag=nonblock)
-DD_OUT_OPTS=(oflag=nonblock conv=fsync)
+DD_OUT_OPTS=(conv=fsync)
```

Comment:
The `oflag=nonblock` flag has been in here since [the initial commit in this repo](https://github.com/endlessm/eos-meta/commit/7590026a55cdc897541abbaefaf6b05305773c2b) but as described in the commit message it causes misleading errors. I can't find the repo that the script was imported from so I do not know if there is some reason why it needs to be there. The [eos-factory-tools version](https://github.com/endlessm/eos-flash-tools/blob/96d6d26933ffa67029638a5c2936af8f00ecc7fd/eos-usb-reflash/99reflash/shutdown.sh#L39-L51) does not include it.

Reply:
It was introduced a long time ago in https://github.com/endlessm/eos-build/commit/62867c6, but there's no justification for it. Maybe @rshuler remembers since he made that change. IIRC, someone may have suggested that using nonblocking I/O could make things go faster. Howerver, I think we don't actually want nonblocking behavior since we're blocking on `dd` to see if it completed. Like you say, doing all the I/O nonblocking would just produce unusual errors.

I think what _could_ actually make things go faster is `oflag=direct`, which opens the file with `O_DIRECT`, but that's kind of orthogonal to this change. I recall reading someone's blog that this can actually make a big difference, but I can't find that now.

Reply:
That was copied from the previous version of the script in eos-shell-build, which we have since deleted.  While we could try to restore that repo and see its commit history, my guess is that came from suggested syntax I found on the web, and at this point I'm sure you know the syntax issues better than I, so please proceed with what you feel is best.